File: ankor-core/src/main/java/at/irian/ankor/messaging/MessageFactory.java
package at.irian.ankor.messaging;

import at.irian.ankor.action.Action;
import at.irian.ankor.change.Change;
import at.irian.ankor.session.ModelSession;

/**
 * @author Manfred Geiler
 */
public class MessageFactory {
    //private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(MessageFactory.class);

    private final String systemName;
    private final MessageIdGenerator messageIdGenerator;

    public MessageFactory(String systemName, MessageIdGenerator messageIdGenerator) {
        this.systemName = systemName;
        this.messageIdGenerator = messageIdGenerator;
    }

    public Message createActionMessage(ModelSession modelSession, String actionPropertyPath, Action action) {
        return new ActionMessage(systemName, modelSession.getId(), messageIdGenerator.create(), actionPropertyPath, action);
    }

    public Message createChangeMessage(ModelSession modelSession, String changedPropertyPath, Change change) {
        return new ChangeMessage(systemName, modelSession.getId(), messageIdGenerator.create(), changedPropertyPath, change);
    }

    public Message createGlobalActionMessage(Action action) {
        return new ActionMessage(systemName, null, messageIdGenerator.create(), null, action);
    }

}

File: #ANKOR.net/Ankor.Core/Messaging/MessageFactory.cs
using System;
using System.Globalization;
using Ankor.Core.Action;

namespace Ankor.Core.Messaging {
	/// <summary>
	/// Description of MessageFactory.
	/// </summary>
	public class MessageFactory {
		public string SystemName { get; private set; }
		public NextId CreateNextId { get; set; }

		// TODO add modelcontext

		public string ModelContextId { get; set; }

		private long currentId = 1;
		public delegate string NextId();
		
		public MessageFactory(string systemName) {
			this.SystemName = systemName;
			CreateNextId = () => SystemName + "#" + currentId++;
			ModelContextId = "1";
		}

		public Message CreateActionMessage(string actionPropertyPath, AAction action) {
			return new ActionMessage(SystemName, ModelContextId, CreateNextId(), actionPropertyPath, action);
		}

		public Message CreateChangeMessage(string changedPropertyPath, Change.Change change) {
			return new ChangeMessage(SystemName, ModelContextId, CreateNextId(), changedPropertyPath, change);
		}

		public Message CreateGlobalActionMessage(AAction action) {
			return new ActionMessage(SystemName, null, CreateNextId(), null, action);
		}


	}
}
